tabs/tab_corr.py

```
@@ -1,300 +1,342 @@
-    icc_result = reactive.Value(None)   # Stores result from ICC analysis
-    numeric_cols_list = reactive.Value([])  # List of numeric columns
+    @reactive.Calc
+    def current_df() -> Optional[pd.DataFrame]:
+        """Select between original and matched dataset based on user preference."""
+        if is_matched.get() and input.radio_corr_source() == "matched":
+            return df_matched.get()
+        return df.get()
     
-    # ==================== UPDATE NUMERIC COLUMNS ====================
+    @render.ui
+    def ui_title_with_summary():
```

Ruff (0.14.10)

Missing return type annotation for private function `ui_title_with_summary`

(ANN202)